### PR TITLE
AJ-1373 use pet token instead of bearer token for import

### PR DESCRIPTION
**Pull request by calypsomatic**
[AJ-1373](https://broadworkbench.atlassian.net/browse/AJ-1373)
This PR takes the pet token fetched in https://github.com/DataBiosphere/terra-workspace-data-service/pull/363 and passes it as the auth token for an Import Job.

Reminder:

PRs merged into main will not automatically generate a PR in https://github.com/broadinstitute/terra-helmfile to update the WDS image deployed to kubernetes - this action will need to be triggered manually by running the following github action: https://github.com/DataBiosphere/terra-workspace-data-service/actions/workflows/tag.yml. Dont forget to provide a Jira Id when triggering the manual action, if no Jira ID is provided the action will not fully succeed. 

After you manually trigger the github action (and it completes with no errors), you must go to [the terra-helmfile](https://github.com/broadinstitute/terra-helmfile) repo and verify that this generated a PR that merged successfully.

The terra-helmfile PR merge will then generate a PR in [leonardo](https://github.com/DataBiosphere/leonardo).  This will automerge if all tests pass, but if jenkins tests fail it will not; be sure to watch it to ensure it merges. To trigger jenkins retest simply comment on PR with "jenkins retest". 

[AJ-1373]: https://broadworkbench.atlassian.net/browse/AJ-1373?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: service/src/main/java/org/databiosphere/workspacedataservice/service/ImportService.java [hunk 1]
```diff
@@ -74,15 +69,12 @@ public GenericJobServerModel createImport(
         importRequest.getType());
 
     try {
-      // get the user's token from the current request
-      // TODO: this should actually get a pet token for the user, to ensure the token won't time out
-      BearerToken token = TokenContextUtil.getToken();
+      // get a token to execute the job
+      String petToken = samDao.getPetToken();
```
[the remark this answers is not in the document]
Comment (calypsomatic) [reply service/src/main/java/org/databiosphere/workspacedataservice/service/ImportService.java]: Good call. Done.

File: service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java [hunk 2]
```diff
@@ -157,4 +189,25 @@ void failsJobIfSchedulingFails(ImportRequestServerModel.TypeEnum importType) {
...
+    // schedulerDao.schedule(), which returns void, returns successfully
+    doNothing().when(schedulerDao).schedule(any(Schedulable.class));
+    // create instance (in the MockInstanceDao)
+    UUID instanceId = UUID.randomUUID();
+    instanceService.createInstance(instanceId, VERSION);
+    // define the import request
+    URI importUri = URI.create("http://does/not/matter");
+    ImportRequestServerModel importRequest = new ImportRequestServerModel(importType, importUri);
+    // Import will fail without a pet token
+    assertThrows(Exception.class, () -> importService.createImport(instanceId, importRequest));
```
Comment (calypsomatic) [inline service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java:211]: I would like to check that a job is not created, but the jobdao can only get a job if it knows the id - is there another way to check or is this good enough?

File: service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java [hunk 3]
```diff
@@ -40,10 +51,30 @@ class ImportServiceTest {
...
+  @BeforeEach
+  void setUp() throws ApiException {
+
+    // return the mock ResourcesApi from the mock SamClientFactory
+    given(mockSamClientFactory.getResourcesApi(null)).willReturn(mockSamResourcesApi);
+    // Sam permission check will always return true
+    given(mockSamResourcesApi.resourcePermissionV2(anyString(), anyString(), anyString()))
+        .willReturn(true);
+    given(mockSamClientFactory.getGoogleApi(null)).willReturn(mockSamGoogleApi);
+    // Sam permission check will always return true
```
Comment (davidangb) [inline service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java:72]: nit: update code comment, looks like copy/paste

File: service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java (hunk 2 above)
Comment (nawatts) [reply service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java:211]: Could you mock the jobDao and assert that `createJob` was not called?

I don't know what the equivalent would be for Junit, but if I were testing this in JS/Jest, I would do something like:

```
spyOn(jobDao, 'createJob')
importService.createImport(...)
expect(jobDao.createJob).not.ToHaveBeenCalled()
```

**Review by nawatts — APPROVED** [review]

File: service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java (hunk 2 above)
Comment (davidangb) [reply service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java:211]: I haven't tried this myself, but … 

can you use `@SpyBean JobDao jobDao;` to enable spying on jobDao, then verify that `jobDao.createJob()` was called zero times? 

Alternately, you could use `@Autowired NamedParameterJdbcTemplate namedTemplate;` and then write a custom SQL statement via the `namedTemplate` to inspect the rows in `sys_wds.job`

**Review by davidangb — COMMENTED** [review]
looking good, added some suggestions on how to test the thing you wanted to test!

File: service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java [hunk 4]
```diff
@@ -39,11 +52,31 @@ class ImportServiceTest {
...
+  @MockBean SamClientFactory mockSamClientFactory;
+
+  ResourcesApi mockSamResourcesApi = Mockito.mock(ResourcesApi.class);
+  GoogleApi mockSamGoogleApi = Mockito.mock(GoogleApi.class);
 
   private static final String VERSION = "v0.2";
 
+  @BeforeEach
+  void setUp() throws ApiException {
+    //    jobDao = Mockito.spy(jobDao);
```
Comment (nawatts) [inline service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java:67]: Is this left over from development?

Comment (calypsomatic) [reply service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java:67]: ah you caught me

File: service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java [hunk 5]
```diff
@@ -39,11 +52,30 @@ class ImportServiceTest {
 
   @Autowired ImportService importService;
   @Autowired InstanceService instanceService;
-  @Autowired JobDao jobDao;
+  @Autowired @SpyBean JobDao jobDao;
```
Comment (davidangb) [inline service/src/test/java/org/databiosphere/workspacedataservice/service/ImportServiceTest.java:55]: nit: the `@Autowired` here is redundant (but harmless):
```suggestion
@SpyBean JobDao jobDao;
```

**Review by davidangb — APPROVED** [review]
nice! :+1:, one nit inline which is not a blocker.

**Pull request merged**